common/string-keyed.h

```diff
@@ -325,7 +325,7 @@ class basic_map_string_kv : public M
     const_iterator end() const
     {
         auto it = base::end();
-        return (B_IT&)it;
+        return __reinterpret_cast<B_IT>(it);
```

Comment:
why not simply use
`reinterpret_cast<B_IT&>(t)` instead?

Reply:
the ```reinterpret_cast``` keyword doesn't support this use